components/collapsible-panel/collapsible-panel-group.js

```diff
@@ -32,39 +33,28 @@ class CollapsiblePanelGroup extends SkeletonGroupMixin(LitElement) {
 
 	constructor() {
 		super();
-		this._panels = [];
+		this._panels = new SubscriberRegistryController(this, 'collapsible-panel-group');
+		this._spaced = true;
 	}
 
 	render() {
 		const classes = {
-			spaced: this._panels?.[0]?.type !== 'inline',
+			spaced: this._spaced,
 		};
 
-		return html`<slot class="${classMap(classes)}" @slotchange="${this._handleSlotChange}"></slot>`;
-	}
-
-	async _getPanels() {
-		const slot = this.shadowRoot?.querySelector('slot');
-		if (!slot) return;
-
-		return slot.assignedNodes({ flatten: true }).filter((node) => node.nodeType === Node.ELEMENT_NODE);
-	}
-
-	async _handleSlotChange() {
-		this._panels = await this._getPanels();
-		this._updatePanelAttributes();
+		return html`<slot class="${classMap(classes)}"></slot>`;
 	}
 
 	_updatePanelAttributes() {
-		if (!this._panels || this._panels.length === 0) return;
-
-		if (this._panels[0]?.type !== 'inline') return;
-
-		for (const panel of this._panels) {
-			panel._noBottomBorder = true;
-		}
-
-		this._panels[this._panels.length - 1]._noBottomBorder = false;
+		const panels = this.shadowRoot
+			?.querySelector('slot')
+			?.assignedNodes({ flatten: true })
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.tagName === 'D2L-COLLAPSIBLE-PANEL');
```

Comment:
We don't need to change since it's merged, but it looks like we can start using `assignedElements` ([canIUse](https://caniuse.com/mdn-api_htmlslotelement_assignedelements)) eh? It basically just avoids having to do the `Node.ELEMENT_NODE` check.

Reply:
Nice!